### PR TITLE
Prevent Airflow Webserver Reboot for Airflow 2.0+

### PR DESCRIPTION
**Pull request by andriisoldatenko**
resolves https://github.com/astronomer/issues/issues/2123

TODO:
- [x] revert https://github.com/astronomer/airflow-chart/pull/157/files
- [x] If airflowVersion >= 2.0 then use RollingUpdate, else use Recreate.

### CODE REVIEW

File: templates/webserver/webserver-deployment.yaml [hunk 1]
```diff
@@ -17,9 +17,21 @@ metadata:
...
+    # capacity during the rolling update.
+    # You can also use percentage based value instead of integer.
     type: RollingUpdate
     rollingUpdate:
-      maxUnavailable: 1
+      maxSurge: 1
+      maxUnavailable: 0
+    {{- else }}
+    type: Recreate
+    {{- end}}
```
Comment (kaxil) [inline templates/webserver/webserver-deployment.yaml:34]: Can we add the test for this code please

Comment (andriisoldatenko) [reply templates/webserver/webserver-deployment.yaml:34]: working on it right now :) @kaxil

Comment (andriisoldatenko) [reply templates/webserver/webserver-deployment.yaml:34]: @kaxil done in 35273c65349c74dc67d3420eac440c06112413ec

**Review by kaxil — APPROVED** [review]

**Comment by vishwas-astro** [comment]
@schnie please review before merge.

**Review by schnie — APPROVED** [review]

**Comment by andriisoldatenko** [comment]
practically LGTM:
when i deploy new dags
```
NAME                                            READY   STATUS        RESTARTS   AGE
chart-1616695849-postgresql-0                   1/1     Running       0          11m
chart-1616695849-run-airflow-migrations-wfzj2   0/1     Completed     0          10s
chart-1616695849-scheduler-7c669cf95-zdcch      2/2     Terminating   0          11m
chart-1616695849-statsd-7f68948ddb-666g9        1/1     Running       0          11m
chart-1616695849-webserver-68fcfdf96-v6p8v      0/1     Running       0          10s
chart-1616695849-webserver-754585d687-l8x6q     1/1     Running       0          11m
```

```
➜  t_20210325_201403 k get pods
NAME                                            READY   STATUS        RESTARTS   AGE
chart-1616695849-postgresql-0                   1/1     Running       0          12m
chart-1616695849-run-airflow-migrations-wfzj2   0/1     Completed     0          28s
chart-1616695849-scheduler-7f565f6669-4tfgc     1/2     Running       0          16s
chart-1616695849-statsd-7f68948ddb-666g9        1/1     Running       0          12m
chart-1616695849-webserver-68fcfdf96-v6p8v      1/1     Running       0          28s
chart-1616695849-webserver-754585d687-l8x6q     0/1     Terminating   0          12m
```

**Comment by andriisoldatenko** [comment]
i've tested it in kind with whole platform for airflow **1.10.12**:
```
spec:
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      component: webserver
      release: transparent-sky-4140
      tier: airflow
  strategy:
    type: Recreate

```

```
➜  astronomer git:(master) ✗ k get pods -n astronomer-transparent-sky-4140
NAME                                                READY   STATUS              RESTARTS   AGE
transparent-sky-4140-flower-746c6f4598-xt8h5        1/1     Running             0          3m18s
transparent-sky-4140-pgbouncer-6d8c9c97f7-h7xx2     2/2     Running             0          3m18s
transparent-sky-4140-redis-0                        1/1     Running             0          3m18s
transparent-sky-4140-run-airflow-migrations-dxpw9   0/1     ContainerCreating   0          2s
transparent-sky-4140-scheduler-5f7f46d4d9-tshcj     2/2     Terminating         0          3m18s
transparent-sky-4140-statsd-6b9769ddf-f78sp         1/1     Running             0          3m18s
transparent-sky-4140-webserver-68d4b8b99-9lfk5      1/1     Terminating         0          3m17s
transparent-sky-4140-worker-6858bf8464-gsmxg        0/1     Init:0/1            0          2s
transparent-sky-4140-worker-767464cb79-bw8fj        1/1     Running             0          3m18s
```


img

**Comment by andriisoldatenko** [comment]
**Airflow 2.0.0:**
```
$ k get deployments ecliptical-earth-4107-webserver -n astronomer-ecliptical-earth-4107 -o yaml | rg -C5 strategy

  selector:
    matchLabels:
      component: webserver
      release: ecliptical-earth-4107
      tier: airflow
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 0
    type: RollingUpdate
  template:
```

```
➜  astronomer git:(master) ✗ k get pods -n astronomer-ecliptical-earth-4107
NAME                                                 READY   STATUS        RESTARTS   AGE
ecliptical-earth-4107-flower-f48b9585-llwlq          1/1     Running       0          4m31s
ecliptical-earth-4107-pgbouncer-5f45bff5fc-rs6sr     2/2     Running       0          4m31s
ecliptical-earth-4107-redis-0                        1/1     Running       0          4m31s
ecliptical-earth-4107-run-airflow-migrations-w4rn4   1/1     Running       0          8s
ecliptical-earth-4107-scheduler-79c8fc8cbc-2p55n     2/2     Terminating   0          4m31s
ecliptical-earth-4107-scheduler-79c8fc8cbc-g6s68     2/2     Terminating   0          4m31s
ecliptical-earth-4107-statsd-544d54cf57-cns9h        1/1     Running       0          4m31s
ecliptical-earth-4107-webserver-675999b548-kclq8     1/1     Running       0          4m31s
ecliptical-earth-4107-webserver-749cb97b86-cgf6x     0/1     Pending       0          8s
ecliptical-earth-4107-worker-578c757fcd-lrqq9        0/1     Pending       0          8s
ecliptical-earth-4107-worker-6dbffc79db-mnbbn        1/1     Running       0          4m31s
```

and i can't see message Airflow is starting up 🎉

**Pull request merged**